packages/core/src/components/side-pane/side-pane.tsx

```diff
@@ -0,0 +1,369 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+import anime from 'animejs';
+import { applicationLayoutService } from '../utils/application-layout';
+import { Breakpoint, matchBreakpoint } from '../utils/breakpoints';
+import Animation from "../utils/animation";
+
+export type SidePanelPosition = 'top' | 'left' | 'bottom' | 'right';
+export type ExpandedChangeEvent = {
+  position: string;
+  expanded: boolean;
+};
+
+@Component({
+  tag: 'ix-side-pane',
+  styleUrl: 'side-pane.scss',
+  shadow: true,
+})
+export class SidePane {
+  @Element() hostElement: HTMLIxSidePaneElement;
+
+  /**
+   * Title of the side panel
+   */
+  @Prop() paneTitle: string = 'Pane title';
+
+  /**
+   * Floating or inline style
+   */
+  @Prop() inline: boolean = false;
+
+  /**
+   * Floating or inline style
+   */
+  @Prop() floating: boolean = false;
+
+  /**
+   * The maximum size of the sidebar, when it is expanded
+   */
+  @Prop() expandedPaneSize:
+    | '240px'
+    | '320px'
+    | '360px'
+    | '480px'
+    | '600px'
+    | '33%'
+    | '50%'
+    | '100%' = '240px';
+
+  /**
+   * Toggle either the preview content is shown or not
+   */
+  @Prop() showPreviewContent: boolean = false;
+
+  /**
+   * State of the side-pane
+   */
+  @Prop({ mutable: true }) expandPane: boolean = false;
+
+  /**
+   * Placement of the sidebar
+   */
+  @Prop({ mutable: true }) position: SidePanelPosition = 'top';
+
+  /**
+   * Supported layouts
+   */
+  @Prop() breakpoints: Breakpoint[] = ['sm', 'md'];
+
+  /**
+   * Event
+   */
+  @Event() expandedChange: EventEmitter<ExpandedChangeEvent>;
```

Comment:
```suggestion
  @Event() expandPaneChange: EventEmitter<ExpandPaneChangeEvent>;
```

- Consistency (new components)
- Angular near naming (https://angular.io/guide/two-way-binding#how-two-way-binding-works)